client/src/main/java/org/asynchttpclient/netty/channel/DefaultChannelPool.java

```diff
@@ -356,6 +357,16 @@ public void flushPartitions(ChannelPoolPartitionSelector selector) {
         }
     }
 
+    @Override
+    public long getIdleChannelCount() {
+        return partitions.reduceValuesToLong(
```

Comment:
This method is neat, but examining the implementation shows that it involves blocking on a `ForkJoinTask`. Unless there's a particular reason to want it, it might be better to just call `partitions.values().stream().map(ConcurrentLinkedDeque::size).count()`.